### PR TITLE
Logo style updates

### PR DESCRIPTION
**Pull request by Ridhwana**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Ensure that the logo shows up properly within the navbar container whether the Feature Flag is enabled or not.

## QA Instructions, Screenshots, Recordings

In your rails console set the following:  

```
Settings::General.original_logo = "https://canary1.forem.wtf/remoteimages/uploads/articles/bcvmn496boszdko7jb09.jpeg"
Settings::General.resized_logo = "https://canary1.forem.wtf/remoteimages/uploads/articles/zn14x1elppv4frwlnbo9.jpeg"

Settings::General.logo_svg = '<svg width="50" height="40" viewBox="0 0 50 40" fill="none" xmlns="http://www.w3.org/2000/svg"><rect width="50" height="40" rx="3" style="fill: currentColor;"></rect><path d="M19.099 23.508c0 1.31-.423 2.388-1.27 3.234-.838.839-1.942 1.258-3.312 1.258h-4.403V12.277h4.492c1.31 0 2.385.423 3.224 1.27.846.838 1.269 1.912 1.269 3.223v6.738zm-2.808 0V16.77c0-.562-.187-.981-.562-1.258-.374-.285-.748-.427-1.122-.427h-1.685v10.107h1.684c.375 0 .75-.138 1.123-.415.375-.285.562-.708.562-1.27zM28.185 28h-5.896c-.562 0-1.03-.187-1.404-.561-.375-.375-.562-.843-.562-1.404V14.243c0-.562.187-1.03.562-1.404.374-.375.842-.562 1.404-.562h5.896v2.808H23.13v3.65h3.088v2.808h-3.088v3.65h5.054V28zm7.12 0c-.936 0-1.684-.655-2.246-1.965l-3.65-13.758h3.089l2.807 10.804 2.808-10.804H41.2l-3.65 13.758C36.99 27.345 36.241 28 35.305 28z" style="fill: var(--base-inverted);"></path></svg>'
```

Go to the home page with the logo and run:
```
FeatureFlag.enable(:creator_onboarding)
```
to see it with the new updates with the  `<img` tag .

Go to the home page with the logo  and run:
```
FeatureFlag.disable(:creator_onboarding)
```
to see it with the the rendered `logo_svg`.

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [ ] Yes
- [x] No, and this is why: style related change
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [x] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/views/layouts/_logo.html.erb [hunk 1]
```diff
@@ -1,8 +1,8 @@
-<a href="/" class="<%= "site-logo" unless FeatureFlag.enabled?(:creator_onboarding) %>" aria-label="<%= t("views.main.aria_home") %>">
-  <% if Settings::General.logo_svg.present? %>
+<a href="<%= root_path %>" class="site-logo" aria-label="<%= t("views.main.aria_home", community: community_name) %>">
+  <% if FeatureFlag.enabled?(:creator_onboarding) %>
+      <img class="site-logo__img" src="<%= Settings::General.resized_logo %>" alt="<%= community_name %>">
```
Comment (citizen428) [inline app/views/layouts/_logo.html.erb:3]: Absolute nitpick but this line seems to be indented too much:

```suggestion
    <img class="site-logo__img" src="<%= Settings::General.resized_logo %>" alt="<%= community_name %>">
```

**Review by citizen428 — APPROVED** [review]

File: app/views/layouts/_logo.html.erb (hunk 1 above)
Comment (ludwiczakpawel) [reply app/views/layouts/_logo.html.erb:3]: whoops!

**Comment by Ridhwana** [comment]
I can't approve it because I created the branch, but it works great! Thank you @ludwiczakpawel!

**Review by ludwiczakpawel — APPROVED** [review]
Hahahah I can approve it! We hacked the system!

image

(please ignore my approval and lets find one more! :D)

File: app/assets/stylesheets/base/main.scss [hunk 2]
```diff
@@ -95,14 +109,24 @@ a {
...
-    width: auto;
-    height: calc(var(--header-height) - var(--su-2) * 2);
-    vertical-align: middle;
-    display: inline-block;
+  // Both focus declarations below are the same but we unfortunately can't
+  // combine them because Safari doesn't recognize it properly.
+  //  • First declaration is for all browsers that support :focus-visible
+  //    which is basically almost all of them
+  //  • Second declaration is for browsers that do not support :focus-visible
+  //    which basically is only Safari right now.
```
Comment (juliannatetreault) [inline app/assets/stylesheets/base/main.scss:120]: This comment! 🙌

**Review by juliannatetreault — APPROVED** [review]

File: app/assets/stylesheets/base/main.scss [hunk 3]
```diff
@@ -69,15 +69,29 @@ a {
...
+  outline: 0;
   overflow-wrap: anywhere;
 
+  &__img,
+  svg {
+    max-width: 100%;
+    max-height: calc(
+      var(--header-height) - var(--su-2) * 2
+    ); // making calculations instead of setting fixed value so should be smooth change if we ever decide to change header's height.
+    height: auto;
```
[the remark this answers is not in the document]
Comment (ludwiczakpawel) [reply app/assets/stylesheets/base/main.scss:88]: Oh yea, I haven't thought about that. That's a neat suggestion.

[the remark this answers is not in the document]
Comment (ludwiczakpawel) [reply app/assets/stylesheets/base/main.scss]: I've just pushed the change. Can you do a sanity check, please?

**Review by nickytonline — APPROVED** [review]

File: app/assets/stylesheets/base/main.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (nickytonline) [reply app/assets/stylesheets/base/main.scss]: Ooops, I think my stuff was cached. Going to recheck.

[the remark this answers is not in the document]
Comment (nickytonline) [reply app/assets/stylesheets/base/main.scss]: Alright, all good.

**Pull request merged**